### ISSUE TITLE
zypper multi-package slowness causing timeouts

### ISSUE THREAD

**Issue by furlongm** [chef/chef#4594]
It seems that aa6663d introduced a new issue. Although the multi-package syntax works now, it seems to constantly loop over the array of packages, sometimes until a timeout occurs.

For an array of packages of length 9 (as below) it will run get_versions 81 times (9 x 9). in zypper.rb. It does this twice, once for get_candidate_versions and once for get_current_versions, so it runs get_versions 162 (81 + 81) times in total for the below array. This means it runs `zypper --non-interactive info package_name` 162 times for 9 packages. It should only need to run 9 times for 9 packages. It seems to pass all packages to get_versions instead of a single package.

```
[2016-02-17T09:30:39+00:00] INFO: Processing zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] action install (net::default line 1)
[2016-02-17T09:30:39+00:00] DEBUG: Providers for generic zypper_package resource enabled on node include: [Chef::Provider::Package::Zypper]
[2016-02-17T09:30:39+00:00] DEBUG: Provider for action install on resource zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is Chef::Provider::Package::Zypper
[2016-02-17T09:30:39+00:00] DEBUG: ethtool
[2016-02-17T09:30:39+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:40+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 4.0-5.2
[2016-02-17T09:30:40+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:40+00:00] DEBUG: tcpdump
[2016-02-17T09:30:40+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:42+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 4.5.1-2.2
[2016-02-17T09:30:42+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:42+00:00] DEBUG: nmap
[2016-02-17T09:30:42+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:43+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 6.47-9.10
[2016-02-17T09:30:43+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:43+00:00] DEBUG: telnet
[2016-02-17T09:30:43+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:44+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 1.2-167.1
[2016-02-17T09:30:44+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:44+00:00] DEBUG: iftop
[2016-02-17T09:30:44+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:45+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 0.99.4-7.2
[2016-02-17T09:30:45+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:45+00:00] DEBUG: whois
[2016-02-17T09:30:45+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:47+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 5.2.10-4.2
[2016-02-17T09:30:47+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:47+00:00] DEBUG: wget
[2016-02-17T09:30:47+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:48+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 1.14-2.2
[2016-02-17T09:30:48+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:48+00:00] DEBUG: iperf
[2016-02-17T09:30:48+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:49+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 3.0.11-3.2
[2016-02-17T09:30:49+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:49+00:00] DEBUG: ipset
[2016-02-17T09:30:49+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:51+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 6.25.1-4.17
[2016-02-17T09:30:51+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:51+00:00] DEBUG: ethtool
[2016-02-17T09:30:51+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:52+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 4.0-5.2
[2016-02-17T09:30:52+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:52+00:00] DEBUG: tcpdump
[2016-02-17T09:30:52+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:53+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 4.5.1-2.2
[2016-02-17T09:30:53+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:53+00:00] DEBUG: nmap
[2016-02-17T09:30:53+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:55+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 6.47-9.10
[2016-02-17T09:30:55+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:55+00:00] DEBUG: telnet
[2016-02-17T09:30:55+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:56+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 1.2-167.1
[2016-02-17T09:30:56+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:56+00:00] DEBUG: iftop
[2016-02-17T09:30:56+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:57+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 0.99.4-7.2
[2016-02-17T09:30:57+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:57+00:00] DEBUG: whois
[2016-02-17T09:30:57+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:30:59+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 5.2.10-4.2
[2016-02-17T09:30:59+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:30:59+00:00] DEBUG: wget
[2016-02-17T09:30:59+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:31:00+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 1.14-2.2
[2016-02-17T09:31:00+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:31:00+00:00] DEBUG: iperf
[2016-02-17T09:31:00+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:31:01+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 3.0.11-3.2
[2016-02-17T09:31:01+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:31:01+00:00] DEBUG: ipset
[2016-02-17T09:31:01+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:31:02+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 6.25.1-4.17
[2016-02-17T09:31:02+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
[2016-02-17T09:31:02+00:00] DEBUG: ethtool
[2016-02-17T09:31:02+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] checking zypper
[2016-02-17T09:31:04+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] version 4.0-5.2
[2016-02-17T09:31:04+00:00] DEBUG: zypper_package[ethtool, tcpdump, nmap, telnet, iftop, whois, wget, iperf, ipset] is installed
... until timeout occurs ...
```

**Comment by lamont-granquist** [comment]
should be already fixed in master by #4591

**Comment by furlongm** [comment]
Confirmed fixed.
